crates/types/src/logs/mod.rs

```diff
@@ -106,28 +109,71 @@ where
     fn prev(self) -> Self;
 }
 
-/// Owned payload.
-#[derive(
-    Debug,
-    Clone,
-    Default,
-    Eq,
-    PartialEq,
-    derive_more::From,
-    derive_more::Into,
-    derive_more::Deref,
-    derive_more::DerefMut,
-)]
-pub struct Payload(Bytes);
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Header {
+    created_at: MillisSinceEpoch,
+    // additional custom headers can be added here. Those should be somewhat
+    // generic and values must be optional.
+    pub custom_data_1: Option<u64>,
```

Comment:
Is this more of an example of how to add new fields or is `custom_data_1` already used?

Reply:
Ah, it's used by my perf. I might come back to this and give it a meaningful name, but I thought to keep it agnostic.